src/Notes/WC_Admin_Notes.php

```diff
@@ -100,6 +103,74 @@ public static function delete_notes_with_name( $name ) {
 		}
 	}
 
+	/**
+	 * Update a note.
+	 *
+	 * @param WC_Admin_Note $note The note that will be deleted.
+	 * @param array         $requested_updates a list of requested updates.
+	 */
+	public static function update_note( $note, $requested_updates ) {
+		$note_changed = false;
+		if ( isset( $requested_updates['status'] ) ) {
+			$note->set_status( $requested_updates['status'] );
+			$note_changed = true;
+		}
+
+		if ( isset( $requested_updates['date_reminder'] ) ) {
+			$note->set_date_reminder( $requested_updates['date_reminder'] );
+			$note_changed = true;
+		}
+
+		if ( isset( $requested_updates['is_deleted'] ) ) {
+			$note->set_is_deleted( $requested_updates['is_deleted'] );
+			$note_changed = true;
+		}
+
+		if ( $note_changed ) {
+			$note->save();
+		}
+	}
+
+	/**
+	 * Soft delete of a note.
+	 *
+	 * @param WC_Admin_Note $note The note that will be deleted.
+	 */
+	public static function delete_note( $note ) {
+		$note->set_is_deleted( 1 );
+		$note->save();
+	}
+
+	/**
+	 * Soft delete of all the admin notes. Returns the deleted items.
+	 *
+	 * @return array Array of notes.
+	 */
+	public static function delete_all_notes() {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+		// Here we filter for the same params we are using to show the note list in client side.
+		$raw_notes = $data_store->get_notes(
+			array(
+				'order'      => 'desc',
+				'orderby'    => 'date_created',
+				'per_page'   => 25,
+				'page'       => 1,
+				'type'       => array( 'info', 'warning' ),
+				'status'     => array( 'unactioned' ),
+				'is_deleted' => 0,
+			)
+		);
+
+		$notes = array();
+		foreach ( (array) $raw_notes as $raw_note ) {
+			$note = new WC_Admin_Note( $raw_note );
+			$note->set_is_deleted( 1 );
```

Comment:
Should we just use the `delete_note()` function above?

Reply:
Yes, you are right, thank you 🙂
This was changed in the commit [7ca0f10](https://github.com/woocommerce/woocommerce-admin/pull/4262/commits/7ca0f1041653ddf11e1cba08a58b0bbdbe1a0bb2)